### ISSUE TITLE
Xteve Lockup and icon URL issues

### ISSUE THREAD

**Issue by GroupMaster** [xteve-project/xTeVe#100]
I am unable to determine the cause, but xteve is "randomly" locking up causing ALL channel requests to be "queued" in a frozen state. Since this particular server is Windows 7, if I were to select the CMD window where xteve is running and hit enter, all the backlogged requests and responses flood by and it goes back online properly serving future requests. This is a BIG problem. The web GUI however seems to still operate during this lockup showing no issues at all.

Also, Channel icons are not being updated upon a change in the m3u source without a complete re-initialization of xteve. To elaborate, Xteve is initially supplied with an m3u with icon URLs pointing to the IPTV provider's source. If the M3U's URLs are updated to a different server/address then refreshed in xteve to pull in the new playlist, the first image URL remains. I am assuming that Xteve's config needs to be removed and rebuilt from scratch for the new URL's to take affect. The refresh/reload does not completely update the Xteve DB as it should.

Lastly, when icon caching is turned on, I have found the Icon URLs to be using 127.0.0.1 as the server. This will not allow the icons to work if Xteve and Plex are on different servers.

- OS : Windows 7
- Arch : Xeon multi-processor 64 bit
- xTeVe Version and Build : 2.1.2 Build: 0121
- Database Version : 2.1.0

Note, the lockup described above also happened in the last several versions. The conditions causing it has not been determined.

**Comment by xteve-project** [comment]
I don't understand your CMD problem. What do you mean by queues? CMD only shows the log.

All URLs will only be updated at the time of your schedule. Or if you save the mapping.
The log then says:
```
2020-02-29 15:15:10 [xTeVe] XEPG:                  Ready to use
```

With image caching, the IP / domain with which xTeVe was last called is always used.
If a client downloads the xteve.m3u via 127.0.0.1, 127.0.0.1 is used. If another client uses 192.168.1.10, this IP is used. When starting xTeVe, the IP is used as the default, which is shown in the log as the web server URL.

**Comment by GroupMaster** [comment]
After being up and operating for however long, you would see the expected log info including the stream requests. At some point all stream requests to xteve begin instantly failing. So I would remote  into the server, windowize the xteve instance and hit the enter key. At that moment all the requests made since it locked up flood by and it begins to operate correctly again.

**Comment by GroupMaster** [comment]
> With image caching, the IP / domain with which xTeVe was last called is always used.
If a client downloads the xteve.m3u via 127.0.0.1, 127.0.0.1 is used. If another client uses 192.168.1.10, this IP is used. When starting xTeVe, the IP is used as the default, which is shown in the log as the web server URL.

The request was made from a different computer on the local network. If what you say were true then a 192.168.x.x address would have been there. I just tested it now and the expected 192 address was used indeed. I don't understand why I got the 127 address before. I know I opened the files on my local workstation directly into notepadd++ using the 192 URL, which 127.0.0.1 wouldn't have worked to begin with because xteve is on a different computer. I really don't know what happened but I did see it.

As far as the image URLs, this is after the image caching was turned off and a new m3u pulled from xteve. The new image URLs were instated in the source m3u file several days before the caching was disabled. I continue to get the wrong image urls in the xteve output.

Basically, I updated some icons and they didn't work so I disabled caching and it still doesn't show the changes.

**Comment by GroupMaster** [comment]
This is happening all the time now and really making a mess of things!

**Comment by xteve-project** [comment]
If image caching doesn't work for you, turn it off. Downloading all images and logos can take a long time after an internet connection and is only active after the next update of all xTeVe files.

From here it is difficult to say what exactly does not work for you.

**Comment by GroupMaster** [comment]
For now I am removing the Windows xteve server and only running it on the debian server where it doesn't have the issue.
